### ISSUE TITLE
Cannot read property 'x' of undefined error for basic example when moving mouse over map

### ISSUE THREAD

**Issue by dmitryrogozhny** [google-map-react/google-map-react#843]
Hi!

I'm getting the "Cannot read property 'x' of undefined" error message for a basic example created in the codesandbox. The error appears when I try to move a mouse cursor over a map.

Here's the link to the codesandbox: https://codesandbox.io/s/google-map-react-error-with-marker-9zwdm

To reproduce the error, remove/comment the `distanceToMouse` property from the map component.

The error message is the same as for #677 issue, but I'm not sure that it's cause by the same problem.

The implementation for the `distanceToMouse` in the codesandbox logs `pt` and `mousePos` values to the console. When moving mouse, a marker object is sometimes `undefined` which causes default implementation of the `distanceToMouse` to fail.

BTW, I've tried the same code in a project created with Create React App and everything works fine (react-scripts version is 3.4.0). 

I don't know why sometimes the `distanceToMouse` function receives `undefined` instead of an object for marker.

The quick workaround is to provide a custom implementation for the `distanceToMouse` function and check for `undefined`.

**Comment by durunvo** [comment]
I have the same issue.
@dmitryrogozhny Good work around though.

**Comment by carloshdelreal** [comment]
I would like to take some time to try to solve this issue, would you mind to give a quick explanation about the implementation and folder structure

**Comment by carloshdelreal** [comment]
# I got a possible solution 

the problem comes in because the `distanceToMouse` function, tries to make a calculation to get a distance, and as the `pt` or the `mousePos` variables are undefined, and the function tries to get the x property, that produces the error, I was browsing to look for the source code and I don't know where to get it, but if you change this code on the `google-map-react/lib/google_maps.js` file:

```javascript
distanceToMouse: function distanceToMouse(pt, mousePos /* , markerProps */) {
    return Math.sqrt((pt.x - mousePos.x) * (pt.x - mousePos.x) + (pt.y - mousePos.y) * (pt.y - mousePos.y));
  }
```
for this one

```javascript
distanceToMouse: function distanceToMouse(pt, mousePos /* , markerProps */) {
    if(pt && mousePos){
      return Math.sqrt((pt.x - mousePos.x) * (pt.x - mousePos.x) + (pt.y - mousePos.y) * (pt.y - mousePos.y));
    }
  }
```
that is enough to get rid of this issue

**Comment by rturk** [comment]
My temporary hacky fix: Disable this function

```
<GoogleMap
 ...
 distanceToMouse={()=>{}}
...
/>
```

**Comment by itsmichaeldiego** [comment]
@dmitryrogozhny @durunvo @rturk @carloshdelreal  Would you guys like to create a PR?

**Comment by rturk** [comment]
I'm open for a PR, however I'm not familiar with the code. 

Based on my quick research I'm curios why the function is defined as a `default` (1) property, looks like this function is rally used

IMO any PR should aim to:

1. Disable this function, looks like is most of the time it will not be used here (1)
1. IF used, improve the code based on @carloshdelreal solution above

(1)
https://github.com/google-map-react/google-map-react/blob/master/src/google_map.js#L145

**Comment by bdell** [comment]
I went ahead and opened up a PR with what I believe is a decent fix for this issue.  I put a much more detailed explanation in the PR but in essence I moved the `MarkerDispatcher` bindings into the `componentDidMount` of `GoogleMapMarkers` so that they are updated once the render function runs.  This is my first PR to this repo so I'm happy to update as needed, but figured I should drop a note on this thread since its what helped me troubleshoot the issue.

cc @itsmichaeldiego @rturk @carloshdelreal @durunvo @dmitryrogozhny
